Cython/Includes/cpython/datetime.pxd

```diff
@@ -76,22 +76,40 @@ cdef extern from "datetime.h":
 
     # Define structure for C API.
     ctypedef struct PyDateTime_CAPI:
+        """
+        #if PY_MAJOR_VERSON < 3 && PY_MINOR_VERSION < 6 && !defined(*DateTime_FromDateAndTimeAndFold)
+        #define DateTime_FromDateAndTimeAndFold(year, month, day, hour, min, sec, usec, tz, fold) \
+        #   PyDateTimeAPI->DateTime_FromDateAndTime(year, month, day, hour, min, sec, usec, tz)
+        #endif
+        #if Py_MAJOR_VERSION < 3 && PY_MINOR_VERSION < 6 && !defined(*Time_FromTimeAndFold)
+        #define PyTime_FromTimeAndFold(hour, minute, second, usecond, fold) \
+        #   PydateTimeAPI->Time_FromTime(hour, minute, second, usecond)
+        #endif
+        """
```

Comment:
I haven't really had time to figure it out completely and test, but thought I'd push a form of "status rapport", is this somewhat what we've been talking about?

Reply:
I think that's (close to) the right idea, although it'll need a bit of tweaking before it works. Most significantly `defined` can only check if a macro exists so you can't use it on a pointer - I think you'll just have to base this on the Python version.

Reply:
>  I think you'll just have to base this on the Python version.

I am not sure if we can do that, the same reason as to why the `.pdx` implements `PyDateTime_CAPI` "manually".